add --find-if-exists to create-space and create-org

cf-bosh opened the pull request:
Add a `--find-if-exists` option to `cf create-space` and `cf create-org`, which will not blow up if a user tries to create a space in an org with a name that already exists, or tries to create an org with a name that already exists.

The motivation is to allow people to write more idempotent scripts that may need to shell out and invoke `cf`.

-- @Amit-PivotalLabs + @jfoley

amitkgupta committed:
add --find-if-exists to create-space

Commit:
Add --find-if-exists to create-org

Also, don't blow up when logging in if no org exists

cf-bosh commented:
Ran the full integration tests locally, they passed, but not every time.  Increased some of the default timeouts by a few seconds and the tests passed with higher regularity.  Travis failures seem unrelated to this PR, indeed the same errors are occurring on some of the recent builds of master.

-- @Amit-PivotalLabs

CardboardMechanic commented:
To me, it also feels strange to expose this feature to users. What is the messaging if the object was found? What if it was created? Do the exit codes differ or is either one considered a success (exit 0)? This is one command trying to do two things and thus introduces a fair amount of complexity to the [already difficult] UX.

Commit:
Increase wait times to improve robustness of some integration/feature specs

Also try to unify some spec style w.r.t expect vs. should, func(args) vs func args, and :key => value vs key: value

mariash commented:
Merged manually https://github.com/cloudfoundry/cf/commit/6e7caffdfb42992d67f3cb7b39a4c4bfe9aa32a3
